backend/server/routers/planner.py

```diff
@@ -51,3 +55,23 @@ def validate_term_planner(plannerData: PlannerData):
     coursesState = validate_terms(data)
 
     return {"courses_state": coursesState}
+
+@router.get("/autoplanning/", response_model=PlannerData)
+def autoplanning(courseCodes: CourseCodes, userData: UserData, programTime: ProgramTime):
+    result = {"program": userData.program, "specialisations": list(userData.specialisations.keys()), "plan": [], "mostRecentPastTerm": {"Y": 0, "T": 0}}
+    courses = [get_course(courseCode) for courseCode in courseCodes.courses]
+    user = User(dict(userData))
+    try:
+        autoplanned = autoplan(courses, user, programTime.startTime, programTime.endTime, programTime.uocMax)
+    except Exception as e:
+        raise HTTPException(status_code=400, detail=f"Error: {e}")
+    
+    for year in range(programTime.startTime[0], programTime.endTime[0] + 1):
+        result["plan"].append([])
+        for term in range(0, 4):
+            result["plan"][year - programTime.startTime[0]].append({})
+            for course in autoplanned:
+                print(course)
+                if course[1][0] == year and course[1][1] == term:
+                    result["plan"][year - programTime.startTime[0]][term][course[0]] = [get_course(course[0]).uoc , None]
```

Comment:
@gtangelo would you like the output in this format? Or would you like it differently?

Reply:
Was there any reason on why we would want to return the UOC and mark alonside the course keys? (I'm assuming the second key is for the marks but feel free to correct me). Can't think anything that we might need that for on the top of my head rn.

Either way, dont think the format matters too much since we can just change it on the FE side anyways.

Though would be good to have it same to the format as `years` field in the `plannerSlice` i.e. something like this?
```
"plan": [
  {"T0":[],"T1":["COMP1511","COMP2511"],"T2":["COMP1521"],"T3":["COMP1531"]},
  {"T0":[],"T1":["COMP3900"],"T2":["COMP2521"],"T3":["COMP3121"]},
  {"T0":[],"T1":["MATH1231"],"T2":["MATH1081"],"T3":["COMP4920"]},
  {"T0":[],"T1":["COMP3131"],"T2":[],"T3":["COMP3311"]}
]
```